### ISSUE TITLE
`bst source-bundle` doesn't work

### ISSUE THREAD

**Issue by cs-shadow** [cs-shadow/buildstream#63]
In GitLab by @sstriker on Aug 4, 2017, 15:32

Running `bst source-bundle my.bst` results in:

```
Traceback (most recent call last):
  File "/usr/local/bin/bst", line 11, in <module>
    load_entry_point('BuildStream==1.0', 'console_scripts', 'bst')()
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/buildstream/_frontend/main.py", line 416, in source_bundle
    compression, except_, directory)
  File "/usr/local/lib/python3.5/dist-packages/buildstream/_pipeline.py", line 763, in source_bundle
    element._write_script(source_directory)
  File "/usr/local/lib/python3.5/dist-packages/buildstream/element.py", line 627, in _write_script
    commands=self.generate_script()
  File "/usr/local/lib/python3.5/dist-packages/buildstream/buildelement.py", line 217, in generate_script
    commands = self.commands[command_name]
AttributeError: 'ImportElement' object has no attribute 'commands'
```

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 10, 2017, 14:31

ImportElement overrides BuildElement's configure method but doesn't override the generate_script method. BuildElement's generate_script method expects the commands attribute to be initialized and otherwise fails with the above exception. This should be easy to fix by also overriding generate_script. There may be further issues in the source-bundle feature, though, which we have to look into.

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 15, 2017, 12:58

mentioned in commit 8e695fe4bacb470cf9095581176451b8bb88083e

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 15, 2017, 12:59

mentioned in merge request !78

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 15, 2017, 12:59

assigned to @juergbi

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 22, 2017, 06:41

mentioned in commit ee52bdf4327578c561ca7b0a1729e12462dcbc16

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 22, 2017, 07:05

closed via commit ee52bdf4327578c561ca7b0a1729e12462dcbc16

**Comment by cs-shadow** [comment]
In GitLab by @juergbi on Aug 22, 2017, 07:05

closed via merge request !78
